omnilib/prototypes/functions/functions-misc.lua

```diff
@@ -823,6 +835,137 @@ function omni.lib.find_entity_prototype(item)
 	return nil
 end
 
+function omni.lib.find_recipe(item)
+	if type(item)=="table" then return item elseif type(item)~="string" then return nil end
+	for _, p in pairs(data.raw.recipe) do
+		if standardized_recipes[p.name] == nil then omni.marathon.standardise(p) end
+		for _,r in pairs(p.normal.results) do
+			if r.name == item then return p end
+		end
+	end
+	--log("Could not find "..item.."'s recipe prototype, check it's type.")
+	return nil
+end
+
+-----------------------------------------------------------------------------
+-- ICON FUNCTIONS --
+-----------------------------------------------------------------------------
+
+--If our prototype doesn't have an icon, we need to find one that does
+local find_icon = function(item)
+	for _, p in pairs({"item","mining-tool","gun","ammo","armor","repair-tool","capsule","module","tool","rail-planner","item-with-entity-data","fluid","technology"}) do
+		if data.raw[p][item] then
+			if data.raw[p][item].icons or data.raw[p][item].icon then
+				return data.raw[p][item]
+			end
+		end
+	end
+end
```

Comment:
this is returning an item prototype, shouldnt be called "find_icon"

Reply:
b116e4a